src/ShopBundle/objects/db/TShopArticleReview.class.php

```diff
@@ -149,27 +149,29 @@ protected function GetAdditionalViewVariables($sViewName, $sViewType)
     }
 
     /**
-     * load data from row, removing extranet user, id, etc.
+     * load data from row, only allowing user-changeable fields.
      *
      * @param array $aRow
      */
     public function LoadFromRowProtected($aRow)
     {
-        $aData = $this->sqlData;
-        if (!is_array($aData)) {
-            $aData = array();
-        }
-        $aInvaliditems = array('id', 'data_extranet_user_id', 'datecreated', 'publish');
-        foreach ($aInvaliditems as $key) {
-            if (array_key_exists($key, $aRow)) {
-                unset($aRow[$key]);
+        $whitelist = $this->getFieldWhitelistForLoadByRow();
+        $safeData = [];
+        foreach ($aRow as $key => $val) {
+            if (\in_array($key, $whitelist, true)) {
+                $safeData[$key] = $val;
             }
         }
-        $oUser = TdbDataExtranetUser::GetInstance();
-        if ($oUser->IsLoggedIn()) {
-            $aRow['data_extranet_user_id'] = $oUser->id;
+        $user = TdbDataExtranetUser::GetInstance();
+        if ($user->IsLoggedIn()) {
+            $safeData['data_extranet_user_id'] = $user->id;
         }
-        $this->LoadFromRow($aRow);
+        $this->LoadFromRow($safeData);
+    }
+
+    protected function getFieldWhitelistForLoadByRow(): array
```

Comment:
How about a simpler `getFieldWhitelist()`? The current name is quite technical, and I'm not sure if the developer is interested in this level of detail when overriding this method (and it could have other uses than in `LoadFromRow()` in the future).

Reply:
LoadFromRowProtected() is quite special. And the method getFieldWhitelistForLoadByRow() belongs to that.
This should be removed altogether.

\TPkgNewsletterUser::getFieldWhitelistForLoadByRow() (old) is also named this way. So this should also be changed there (legacy).

Reply:
Phew OK ..